apps/api/src/app/portfolio/calculator/twr/portfolio-calculator.ts

```diff
@@ -839,14 +844,12 @@ export class TWRPortfolioCalculator extends PortfolioCalculator {
       let average = new Big(0);
       let dayCount = 0;
 
-      if (!this.chartDatesDescending) {
-        this.chartDatesDescending = Object.keys(chartDateMap).sort().reverse();
-      }
+      for (let i = this.chartDates.length - 1; i >= 0; i -= 1) {
```

Comment:
What speaks against this?

```
for (const date of this.chartDates.reverse()) {
```

I think it is easier to understand. Drawback: `reverse()` is destructive

Reply:
This method (`getSymbolMetrics`) is called multiple times during the snapshot computation. So if we reverse the array here, we'd need to reverse it back at the top of the method where we traverse `this.chartDates` in chronological order.

Alternatively, we could cache both `this.chartDates` and `this.chartDatesDescending` (or `this.chartDatesReverse`). The extra memory shouldn't be too bad. What do you think?

Reply:
Thanks for the explanation. I think another variable does not make sense just for that 🤓 Maybe change to `i--` and add a comment:

```
for (let i = this.chartDates.length - 1; i >= 0; i--) {
  // Loop through chart dates in reverse order
  const date = this.chartDates[i];
  ...
}
```